### ISSUE TITLE
Support `padding_side` in `Blip2Processor`

### ISSUE THREAD

**Issue by jemmyshin** [huggingface/transformers#22014]
### Feature request

Support different `padding_side` for `Blip2Processor`.

### Motivation

When I use BLIP2 LLM, I found that the padding style is different between decoder-only model (opt-2.7b for example) and encoder-decoder model (flan-t5-xl for example). So I assume that the paddings are different from `Salesforce/blip2-opt-2.7b` and `Salesforce/blip2-flan-t5-xl`. But actually I got the same padding results, the default is `padding_side`=right. 

Code example:
```
prompt = ["hello world", "Question: how many cats are there? Answer:"]
processor_1 = Blip2Processor.from_pretrained("Salesforce/blip2-opt-2.7b")
processor_2 = Blip2Processor.from_pretrained("Salesforce/blip2-flan-t5-xl")

inputs_1 = processor_1(text=prompt, return_tensors="pt", padding=True)
inputs_2 = processor_2(text=prompt, return_tensors="pt", padding=True)
```

Output (the same for inputs_1 and inputs_2):
```
{'input_ids': tensor([[    2, 42891,   232,     1,     1,     1,     1,     1,     1,     1,
             1],
        [    2, 45641,    35,   141,   171, 10017,    32,    89,   116, 31652,
            35]]), 'attention_mask': tensor([[1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0],
        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]])}
```

I believe that right padding for flan-t5 will give the wrong outputs when calling `generate`, correct me if I am wrong: (transformers/generation/utils.py)
<img width="940" alt="image" src="https://user-images.githubusercontent.com/16580382/223613908-3ca57575-5536-4296-a449-e49ad3b4fa90.png">

Expected outputs (when setting `padding_side`=left):
```
{'input_ids': tensor([[    1,     1,     1,     1,     1,     1,     1,
             1,     2, 42891,   232],
        [    2, 45641,    35,   141,   171, 10017,    32,    89,   116, 31652,
            35]]), 'attention_mask': tensor([[0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1],
        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]])}
```

### Your contribution

I found the this `padding_side` feature exists in `AutoTokenizer`, is that possible to move this feature in `Blip2Processor`?

**Comment by NielsRogge** [comment]
Hi,

You can achieve that by simply updating the padding_side attribute of the processor's tokenizer:
```
processor.tokenizer.padding_side = "left"
```
Note that Blip2Processor is just a wrapper around both the image processor and the tokenizer.

**Comment by jemmyshin** [comment]
Thanks a lot!
